### PR TITLE
Berlin

### PR DESCRIPTION
**Pull request by rachit77**
# Description

This PR adds the functionality of EIP-2718 and EIP-2930 in edge client.
Reference: https://eips.ethereum.org/EIPS/eip-2718 and https://eips.ethereum.org/EIPS/eip-2930

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: chain/params.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ const (
 	QuorumCalcAlignment = "quorumcalcalignment"
 	TxHashWithType      = "txHashWithType"
 	EIP2929             = "EIP2929"
+	EIP2930             = "EIP2930"
 	EIP2565             = "EIP2565"
```
Comment (Stefan-Ethernal) [inline chain/params.go]: I think we should merge these forks into a single, `berlin`, fork. WDYT?

File: types/access_list_tx.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	ChainID    *big.Int
+	AccessList TxAccessList
+}
+
+func (tx *AccessListStruct) transactionType() TxType { return AccessListTx }
+func (tx *AccessListStruct) chainID() *big.Int       { return tx.ChainID }
+func (tx *AccessListStruct) input() []byte           { return tx.Input }
+func (tx *AccessListStruct) gas() uint64             { return tx.Gas }
+func (tx *AccessListStruct) gasPrice() *big.Int      { return tx.GasPrice } //check:EIP2718
```
Comment (Stefan-Ethernal) [inline types/access_list_tx.go:44]: What does this comment mean? Is there a pending work to be done?

File: chain/params.go (hunk 1 above)
Comment (rachit77) [reply chain/params.go]: yeah I will do that

File: types/access_list_tx.go (hunk 2 above)
Comment (rachit77) [reply types/access_list_tx.go:44]: Nope, missed that comment there. I will remove that.

**Review by jelacamarko — APPROVED** [review]

**Comment by begmaroman** [comment]
It is quite difficult to review such a huge PR at once. I had the same problem when implementing London hardfork. What do you think about splitting this PR into multiple separate ones that would have the same parent branch?

**Comment by vcastellm** [comment]
Code freeze

**Pull request closed without merging**